goudpscan/scanner.go

```diff
@@ -127,26 +124,31 @@ func BreakUPIP(segments []string) []string {
 		subnets = append(subnets, head)
 	}
 
-	return subnets
+	return subnets, nil
 }
 
-func ParseSubnet(subnet string) []string {
+func ParseSubnet(subnet string) ([]string, error) {
 	segments := Segmentation(subnet)
-	subnets := []string{}
 
 	if strings.Contains(subnet, "/") {
-		subnets = BreakUPIP(segments[:len(segments)-1])
+		subnets, err := BreakUPIP(segments[:len(segments)-1])
+		if err != nil {
+			return nil, fmt.Errorf("breaking UPIP: %w", err)
+		}
 		for i := 0; i < len(subnets); i++ {
 			subnets[i] = subnets[i] + "/" + segments[len(segments)-1]
 		}
-	} else {
-		subnets = BreakUPIP(segments)
 	}
 
-	return subnets
+	subnets, err := BreakUPIP(segments)
+	if err != nil {
+		return nil, fmt.Errorf("breaking UPIP: %w", err)
+	}
```

Comment:
The logic is broken here, we should only parse the whole `segments` array if there's no network definition "/"

Reply:
Fixed ✅